cedar-policy-core/src/evaluator.rs

```diff
@@ -758,16 +756,9 @@ impl<'e> Evaluator<'e> {
                 value: ValueKind::Lit(Literal::EntityUID(uid)),
                 loc,
             }) => match self.entities.entity(uid.as_ref()) {
-                Dereference::NoSuchEntity => Err(match *uid.entity_type() {
-                    EntityType::Unspecified => EvaluationError::unspecified_entity_access(
-                        attr.clone(),
-                        source_loc.cloned(),
-                    ),
-                    EntityType::Specified(_) => {
-                        // intentionally using the location of the euid (the LHS) and not the entire GetAttr expression
-                        EvaluationError::entity_does_not_exist(uid.clone(), loc)
-                    }
-                }),
+                Dereference::NoSuchEntity => {
+                    Err(EvaluationError::entity_does_not_exist(uid.clone(), loc))
```

Comment:
Saving a comment from deletion
```suggestion
                    // intentionally using the location of the euid (the LHS) and not the entire GetAttr expression
                    Err(EvaluationError::entity_does_not_exist(uid.clone(), loc))
```